### PR TITLE
Enable install of Preview MSI release side-by-side with Stable release

### LINKED ISSUE #8289: PowerShell-6.2.0-preview.2-win-x64 error on startup (also occurs on upgrade to preview.3)
Steps to reproduce
------------------
pwsh-preview

Expected behavior
-----------------
Launch PowerShell-6.2.0-preview.2

Actual behavior
---------------
Gives the following error:

> Error:
>   An assembly specified in the application dependencies manifest (pwsh.deps.json) was not found:
>     package: 'System.Private.ServiceModel', version: '4.5.3'
>     path: 'runtimes/win/lib/netstandard2.0/System.Private.ServiceModel.dll'

Environment data
----------------

```powershell
> $PSVersionTable

```
N/A

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

When installing (or upgrading) Preview builds the component ids for the files are the same as the installed files from the stable MSI.  MSI sees this and assumes the file is already installed so skips installing files that haven't changed (been updated) which results in missing files in the preview install folder and pwsh fails to start.  Fix is to dynamically generate new component ids (and compoentrefs) in `files.wxs` using `_Preview` suffix and use that when building a preview package.

Tested manually.

cc @bergmeister if you can help validate different scenarios.

Fix https://github.com/PowerShell/PowerShell/issues/8289

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]
LGTM

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -2636,6 +2644,36 @@ function New-MSIPackage
...
+    {
+        $component.Id = $component.Id + "_Preview"
+    }
+
+    foreach($componentRef in $filesAssetXml.GetElementsByTagName('ComponentRef'))
+    {
+        $componentRef.Id = $componentRef.Id + "_Preview"
+    }
+
+    $previewFilesWxsPath = Join-Path ([System.IO.Path]::GetTempPath()) "Files-Preview.wxs"
```
Comment (bergmeister) [inline tools/packaging/packaging.psm1]: As far as I can see, this file does not get cleaned up at the end

**Review by bergmeister — COMMENTED** [review]
Interesting approach, at first it seems to be the right approach but then I was thinking to myself:
What if we just used `"*"` (meaning: auto-generate random GUID) for the Component GUIDs in `Files.wxs`, would that not achieve the same without needing to write code?
In general: what about applying something similar to `Product.wxs` as well whatever the best approach turns out to be?

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (SteveL-MSFT) [reply tools/packaging/packaging.psm1]: Good catch.  Will fix.

**Comment by SteveL-MSFT** [comment]
@bergmeister If I'm understanding the docs correctly:

> Component identifier; this is the primary key for identifying components. If omitted, the compiler defaults the identifier to the identifier of the resource that is the explicit keypath of the component (for example, a child File element with KeyPath attribute with value 'yes'.

The generated id is the keypath of the file: `Source="$(env.ProductSourcePath)\System.DirectoryServices.dll"`

`$env:ProductSourcePath` is the source of the built binaries, not the destination.  So it'd be exactly the same for stable and preview builds.

**Comment by bergmeister** [comment]
Thanks Steve for clarification and cleaning up the temporary file. I noticed that the deletion of temporary files does happen using the usual try-finally in this function but I think for the purpose of this function that does not get called too often, it's probably good enough for that.
Looks good to me otherwise.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**